src/main/java/blusunrize/immersiveengineering/client/models/smart/ConnModelReal.java

```diff
@@ -183,16 +183,22 @@ public boolean equals(Object obj)
 			if (!(obj instanceof ExtBlockstateAdapter))
 				return false;
 			ExtBlockstateAdapter o = (ExtBlockstateAdapter) obj;
-			for (IProperty i : state.getPropertyNames())
+			for (IProperty<?> i : state.getPropertyNames())
 			{
+				if (!o.state.getProperties().containsKey(i))
+					return false;
 				Object valOther = o.state.getValue(i);
-				if (valOther == null || !valOther.equals(state.getValue(i)))
+				Object valMine = o.state.getValue(i);
```

Comment:
Yeah I am aware of this issue since i have addressed it for 1.9 and 1.10. I'll see too it, I hope to get the last 1.8 build out this weekend and go full up 1.9/1.10 after that.